OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/ReflectionRelatedFieldAccessesAnalysis.scala

```diff
@@ -0,0 +1,1055 @@
+                project,
+                () => failure(accessPC, actualReceiver, actualParameter, currentMatchers),
+                onlyFieldsExactlyInClass,
+            )
+        }
+
+        var matchers = baseMatchers
+        stmts(fieldDefSite).asAssignment.expr match {
+            case call @ VirtualFunctionCall(_, ObjectType.Class, _, "getDeclaredField" | "getField", _, receiver, params) =>
+                val isGetField = call.name == "getField"
+                if (isGetField) matchers += PublicFieldMatcher
+
+                if (!matchers.contains(NoFieldsMatcher))
+                    matchers += MatcherUtil.retrieveNameBasedFieldMatcher(
+                        context,
+                        params.head.asVar,
+                        NameDepender(accessPC, actualReceiver, actualParameter, matchers, params.head.asVar, stmts, receiver.asVar, context),
+                        accessPC,
+                        stmts,
+                        () => failure(accessPC, actualReceiver, actualParameter, matchers)
+                    )
+
+                if (!matchers.contains(NoFieldsMatcher))
+                    matchers += constructClassBasedFieldMatcherForClassVar(receiver.asVar, matchers, !isGetField)
+
+            case call @ VirtualFunctionCall(_, ObjectType.Class, _, "getDeclaredFields" | "getFields", _, receiver, _) =>
+                val isGetFields = call.name == "getFields"
+                if (isGetFields)
+                    matchers += PublicFieldMatcher
+                if (!matchers.contains(NoFieldsMatcher))
+                    matchers += constructClassBasedFieldMatcherForClassVar(receiver.asVar, matchers, !isGetFields)
+
+            case ArrayLoad(_, _, arrayRef) =>
+                val arrayDepender = ArrayDepender(accessPC, actualReceiver, actualParameter, matchers, arrayRef.asVar, stmts)
+
+                AllocationsUtil.handleAllocations(
+                    arrayRef.asVar, context, arrayDepender, stmts, _ eq ObjectType.Field,
+                    () => failure(accessPC, actualReceiver, actualParameter, baseMatchers)
+                ) { (allocationContext, allocationIndex, stmts) =>
+                        matchers ++= handleGetField(
+                            allocationContext, accessPC, allocationIndex,
+                            actualReceiver, actualParameter,
+                            baseMatchers, stmts
+                        )
+                    }
+
+            case _ =>
+                if (HighSoundnessMode) {
+                    matchers += AllFieldsMatcher
+                } else {
+                    indirectFieldAccesses.addIncompleteAccessSite(accessPC)
+                    matchers += NoFieldsMatcher
+                }
+        }
+
+        matchers
+    }
+}
+
+class FieldGetAnalysis private[analyses] (
+        final val project:       SomeProject,
+        final val apiMethodName: String,
+        final val fieldType:     Option[BaseType] = None
```

Comment:
Since this is only ever used with the default argument value once, maybe remove the default and supply it explicitly in that single case as well for better clarity.

Reply:
I am against this, as it deteriorates two things:
1. It removes the hard typing support that every explicitly given type must be a `BaseType`
2. It makes us add a `UBTypeBasedFieldMatcher` always instead of just when a type is explicitly given

Point 2 can be workarounded (not really asthetically pleasing but its okay), but trying to fix Point 1 would necessitate usage like the `Either` Type which is again verbose and probably not what you want. Dropping Type support is also not what we want.

Reply:
I don't understand what you are saying here. Why would removing the "None" here and putting it explicitly in the one place that uses this constructor without an explicit third parameter (i.e., line 981) have these consequences?

Reply:
Apparently i misunderstood... I thought you wanted to give the types directly without wrapping in an `Option`. By "the default" i thought you referred to the `ObjectType.Object` down below given as the default for the API method return type, as opposed to the "None". Will implement.